src/neptune/common/backends/utils.py

```diff
@@ -95,12 +96,23 @@ def wrapper(*args, **kwargs):
                 HTTPServiceUnavailable,
                 HTTPGatewayTimeout,
                 HTTPBadGateway,
-                HTTPTooManyRequests,
                 HTTPInternalServerError,
                 NewConnectionError,
                 ChunkedEncodingError,
             ) as e:
-                time.sleep(min(2 ** min(10, retry), MAX_RETRY_TIME))
+                time.sleep(min(2 ** min(MAX_RETRY_MULTIPLIER, retry), MAX_RETRY_TIME))
+                last_exception = e
+                continue
+            except HTTPTooManyRequests as e:
+                try:
+                    wait_time = (
```

Comment:
How about extraction of this logic into separate function? Maybe even something in `neptune.common` to be able to use it in both legacy and new client.

Reply:
Sounds very good, I actually put it in utils, if this is okay, cause it is a very specific use case! If you think common is a better place, lmk!

Reply:
Sounds very good, I actually put it in utils, if this is okay, cause it is a very specific use case! If you think common is a better place, lmk!

Reply:
Sounds very good, I actually put it in utils, if this is okay, cause it is a very specific use case! If you think common is a better place, lmk!

Reply:
Ohh, I now realized it was already in `common` module. Let's then leave it just as a function inside of the same file (`src/neptune/common/backends.utils.py`) especially that you had to redefine the constants. Let's just reuse it in legacy client.